parser_library/src/parsing/parser_impl.cpp

```diff
@@ -267,6 +267,22 @@ bool parser_impl::ALIAS()
     return opcode.type == instruction_type::ASM && opcode.value == hlasm_ctx->ids().well_known.ALIAS;
 }
 
+bool parser_impl::NOT(const antlr4::Token* token)
+{
+    if (!token)
+        return false;
+
+    static constexpr std::string_view NOT_OPERATOR = "NOT";
+    auto token_txt = token->getText();
+    if (token_txt.size() != NOT_OPERATOR.size())
+        return false;
+
+    return std::equal(
+        NOT_OPERATOR.begin(), NOT_OPERATOR.end(), token_txt.begin(), [](unsigned char l, unsigned char r) {
+            return l == ::toupper(r);
+        });
```

Comment:
if
```suggestion
    return std::equal(
        NOT_OPERATOR.begin(), NOT_OPERATOR.end(), token_txt.begin(), token_txt.end(), [](unsigned char l, unsigned char r) {
            return l == ::toupper(r);
        });
```
then the extra length check can be droped.